modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/EmptyBucketUserAssignmentForBatchAssignment.java

```diff
@@ -0,0 +1,212 @@
+import com.intuit.wasabi.tests.library.util.Constants;
+import com.intuit.wasabi.tests.model.Application;
+import com.intuit.wasabi.tests.model.Assignment;
+import com.intuit.wasabi.tests.model.Bucket;
+import com.intuit.wasabi.tests.model.Experiment;
+import com.intuit.wasabi.tests.model.User;
+import com.intuit.wasabi.tests.model.factory.ApplicationFactory;
+import com.intuit.wasabi.tests.model.factory.BucketFactory;
+import com.intuit.wasabi.tests.model.factory.ExperimentFactory;
+
+/**
+ * Created on 1st march 2017
+ * This test class tests scenario where a user is assigned to a batch of experiments
+ * and the assigned buckets are deleted to verify that when the assignment call is made
+ * to the same user, then the user should be assigned to one of the available buckets 
+ * in each of the experiments... to keep things simple creating two experiments with
+ * two buckets for each experiment 
+ *
+ */
+public class EmptyBucketUserAssignmentForBatchAssignment extends TestBase {
+
+    Application batchAssignmentEmptyBucketTestApp = null; // app under which the test experiments will be created
+
+    List<Experiment> testExperimentsList = null; // holds experiments that are created in the test class to be deleted eventually in the tearDown
+
+    List<String> assignedBucketLabels = null; // holds the labels of the buckets to which the test user has been assigned
+
+    @BeforeClass
+    public void setUp()
+    {
+        //lets create application and assign it to the variable
+        batchAssignmentEmptyBucketTestApp = ApplicationFactory.createApplication().setName("batchAssignmentEmptyBucketTestApp");
+
+
+        // initialize the declared reference variable 
+       
+        testExperimentsList = new ArrayList<Experiment>();
+
+    }
+
+    /**
+     * This test case tests scenario where we do batch assignment of a user to
+     * list of experiments, delete the assigned buckets and do assignment again
+     * for the same user against the same experiments should make a new assignment
+     */
+    @Test
+    public void createBatchAssignmentAndDeleteAssignedBucketTest() {
+
+        List<Experiment> expList = new ArrayList<Experiment>();
+        assignedBucketLabels = new ArrayList<String>();
+        for(int i = 1 ; i <= 2 ; i++) {
+            Experiment exp = ExperimentFactory.createExperiment("_"+UUID.randomUUID().toString(),-1).setApplication(batchAssignmentEmptyBucketTestApp);
+            exp = postExperiment(exp);
+
+            //create buckets for the experiment
+            List<Bucket> bucketList = BucketFactory.createBuckets(exp, 2);
+            postBuckets(bucketList);
+
+            //below we are starting the experiment
+            exp.state = Constants.EXPERIMENT_STATE_RUNNING;
+            putExperiment(exp);
+
+            testExperimentsList.add(exp);
+            expList.add(exp);
+        }
+
+        // do a POST of assignments for user1234 with context set as PROD
```

Comment:
'context' is deprecated (although, we haven't called it explicitly, it is) and at some point in future we will be removing it, so I would think for any new tests should be doing without 'context'